tests/5.0/target/test_target_collapse.c

```diff
@@ -0,0 +1,63 @@
+//===------ test_target_collapse.c --------------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+// Tests the target directive with collapse clause.
+//
+//===------------------------------------------------------------------------===//
+
+
+#include <stdio.h>
+#include "omp.h"
+#include "ompvv.h"
+
+#define N 5
+int Runtst(int gpu) {
+  omp_set_num_threads(N);
```

Comment:
What is the reason to set the number of threads to N?, Given that the test is checking the collapse. So, why would it fails if for some reason the number of available threads is less than N?
My suggestion is to remove everything related to the number of threads (do not make it fail, based on the number of threads used by the runtime). I ran the test on frontier, printing the values obtained by ThrdTrack, and the number of threads used where different for each GPU. I got
```
For the GPU 0:
ThrdTrack[0]=0
ThrdTrack[1]=1
ThrdTrack[2]=2
ThrdTrack[3]=3
ThrdTrack[4]=4
ThrdTrack[5]=5
ThrdTrack[6]=6
ThrdTrack[7]=7
ThrdTrack[8]=8
ThrdTrack[9]=9
ThrdTrack[10]=10
ThrdTrack[11]=11
ThrdTrack[12]=12
ThrdTrack[13]=13
ThrdTrack[14]=14
ThrdTrack[15]=15
ThrdTrack[16]=16
ThrdTrack[17]=17
ThrdTrack[18]=18
ThrdTrack[19]=19
ThrdTrack[20]=20
ThrdTrack[21]=21
ThrdTrack[22]=22
ThrdTrack[23]=23
ThrdTrack[24]=24
ThrdTrack[25]=0
ThrdTrack[26]=0
ThrdTrack[27]=0
ThrdTrack[28]=0
ThrdTrack[29]=0
For the GPU 1:
ThrdTrack[0]=0
ThrdTrack[1]=1
ThrdTrack[2]=2
ThrdTrack[3]=3
ThrdTrack[4]=4
ThrdTrack[5]=0
ThrdTrack[6]=0
ThrdTrack[7]=0
ThrdTrack[8]=0
ThrdTrack[9]=0
ThrdTrack[10]=0
ThrdTrack[11]=0
ThrdTrack[12]=0
ThrdTrack[13]=0
ThrdTrack[14]=0
ThrdTrack[15]=0
ThrdTrack[16]=0
ThrdTrack[17]=0
ThrdTrack[18]=0
ThrdTrack[19]=0
ThrdTrack[20]=0
ThrdTrack[21]=0
ThrdTrack[22]=0
ThrdTrack[23]=0
ThrdTrack[24]=0
ThrdTrack[25]=0
ThrdTrack[26]=0
ThrdTrack[27]=0
ThrdTrack[28]=0
ThrdTrack[29]=0

```